craft_cli/messages.py

```diff
@@ -61,6 +63,9 @@ class _MessageInfo:  # pylint: disable=too-many-instance-attributes
 # seconds between each spinner char
 _SPINNER_DELAY = 0.1
 
+# the size of bytes chunk that the pipe reader will read at once
+_PIPE_READER_CHUNK_SIZE = 4096
```

Comment:
Shouldn't we make it line-oriented instead of chunk-oriented, since this will be used for text and not for binary data?

Reply:
But you can't.

You have a pipe, and you read bytes from the pipe. You have to read chunks from the pipe (reading byte by byte is too inefficient). And you don't know when in the future you will receive a new line. 

Furthermore, you can not interpret this as text as can not decode the bytes until you're sure you don't have only parts of the bytes; IOW, don't fall in something like the following:

```
>>> "moño".encode("utf8")[:3]
b'mo\xc3'
>>> _.decode("utf8")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc3 in position 2: unexpected end of data
``` 

So it reads chunks of bytes, and then YES, it builds **lines** and those lines are dcoded and sent to the printer (which is line-oriented).

Reply:
Ah yes, there's the encoding problem. We may want to adjust the buffer size however, but let's see how this behaves in a real use case.